.github/workflows/publish_lib.yml

```
@@ -0,0 +1,39 @@
+#
+# Copyright 2023 ABSA Group Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+name: Release - publish client library
+on:
+  push:
+    branches: [master]
+    tags: ["*"]
```

```suggestion
on:
  push:
    branches:
      - 'master'
      - 'support/**'
    tags: 
      - 'v[0-9]+.[0-9]+.[0-9]+*'
```

We may need to release hotfixes that do not get merged to `master` and will reside in the a support branch.

---

Ok, I think the release version works so we don't create new versions on every push to master as we may be doing changes that don't affect the library at all, so a new version getting pushed to Maven would be pointless.